tests/test_create.py

```diff
@@ -753,17 +761,16 @@ def test_install_tarball_from_local_channel(clear_package_cache: None):
 
         # Regression test for 2970
         # install from build channel as a tarball
-        tar_path = join(PackageCacheData.first_writable().pkgs_dir, flask_fname)
-        if not os.path.isfile(tar_path):
-            tar_path = tar_path.replace(".conda", ".tar.bz2")
-        conda_bld = join(
-            dirname(PackageCacheData.first_writable().pkgs_dir), "conda-bld"
-        )
-        conda_bld_sub = join(conda_bld, context.subdir)
-        if not isdir(conda_bld_sub):
-            os.makedirs(conda_bld_sub)
-        tar_bld_path = join(conda_bld_sub, basename(tar_path))
+        tar_path = Path(PackageCacheData.first_writable().pkgs_dir, flask_fname)
+        if not tar_path.is_file():
+            tar_path = tar_path.with_suffix(".tar.bz2")
+
+        # create a temporary conda-bld
+        conda_bld_sub = tmp_path / context.subdir
+        conda_bld_sub.mkdir(exist_ok=True)
+        tar_bld_path = str(conda_bld_sub / tar_path.name)
```

Comment:
We move the tarball into a temporary directory to avoid polluting subsequent tests.